build/mac/notarize_dmg_pkg.py

```diff
@@ -0,0 +1,186 @@
+# relative to that
+packaging_signing_path = os.path.realpath(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(packaging_signing_path)
+
+# Import the entire module to avoid circular dependencies in the functions
+from signing import config, commands, model, notarize, pipeline, signing  # noqa: E402
+
+
+def run_command(args, **kwargs):
+    print('Running command: {}'.format(args))
+    subprocess.check_call(args, **kwargs)
+
+
+def create_config(config_args, development, mac_provisioning_profile):
+    """Creates the |model.CodeSignConfig| for the signing operations.
+
+    If |development| is True, the config will be modified to not require
+    restricted internal assets, nor will the products be required to match
+    specific certificate hashes.
+
+    Args:
+        config_args: List of args to expand to the config class's constructor.
+        development: Boolean indicating whether or not to modify the chosen
+            config for development testing.
+
+    Returns:
+        An instance of |model.CodeSignConfig|.
+    """
+    config_class = config.CodeSignConfig
+    """
+    try:
+        import signing.internal_config
+        config_class = signing.internal_config.InternalCodeSignConfig
+    except ImportError as e:
+        # If the build specified Google Chrome as the product, then the
+        # internal config has to be available.
+        if config_class(identity, keychain).product == 'Google Chrome':
+            raise e
+    """
+
+    if development:
+
+        class DevelopmentCodeSignConfig(config_class):
+
+            @property
+            def codesign_requirements_basic(self):
+                return ''
+
+            @property
+            def provisioning_profile_basename(self):
+                return None
+
+            @property
+            def run_spctl_assess(self):
+                return False
+
+        config_class = DevelopmentCodeSignConfig
+
+    config_class = GetBraveSigningConfig(config_class, development, mac_provisioning_profile)
+    return config_class(*config_args)
+
+
+def GetBraveSigningConfig(config_class, development, mac_provisioning_profile):
+    if development:
+        return config_class
+
+    if mac_provisioning_profile:
+        provisioning_profile = mac_provisioning_profile
+        assert len(provisioning_profile), 'Argument mac_provisioning_profile not provided!'
+
+    class ProvisioningProfileCodeSignConfig(config_class):
+
+        @property
+        def provisioning_profile_basename(self):
+            return os.path.splitext(os.path.basename(
+                provisioning_profile))[0]
+
+        @property
+        def run_spctl_assess(self):
+            return True
+
+    return ProvisioningProfileCodeSignConfig
+
+
+def NotarizeBraveDmgPkg(paths, config, dmg, pkg, outdir, signed, do_notarization=True):
+    """
+    Notarize Brave .dmg and .pkg files.
+    """
+    if do_notarization:
```

Comment:
Seems like this can be checked in main before creating configs.

Reply:
This was copied from the chromium `signing/pipeline.py` code and is unnecessary here. Removing.